src/components/AssetItem/AssetItem.js

```diff
@@ -0,0 +1,178 @@
+// @flow
+
+import React, { PureComponent, Fragment } from 'react'
+
+import classNames from 'classnames'
+import { handle } from 'utils/eventHandlers'
+import AssetBalance from 'components/AssetBalance'
+import { JAssetSymbol, JText, JCard, JIcon, JFlatButton } from 'components/base'
+
+type Props = {
+  deleteAssetItem: () => void,
+  +name: string,
+  +symbol: string,
+  +balance: number,
+  +isCustom: boolean,
+}
+
+type StateProps = {|
+  isChange: boolean,
+  isHoveredEdit: boolean,
+  isHoveredTrash: boolean,
+  isDeleteDialog: boolean,
+|}
+
+class AssetItem extends PureComponent<Props, StateProps> {
+  static defaultProps = {
+    fiatBalance: 0,
+    balance: 0,
+    isCustom: false,
+  }
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      isChange: false,
+      isHoveredEdit: false,
+      isHoveredTrash: false,
+      isDeleteDialog: false,
+    }
+  }
+
+  onHoverEdit = (isHoveredEdit: boolean) => {
+    this.setState({ isHoveredEdit })
+  }
+
+  onHoverTrash = (isHoveredTrash: boolean) => {
+    this.setState({ isHoveredTrash })
+  }
+
+  toggleChange = (isChange: boolean) => {
+    this.setState({
+      isChange,
+    })
+  }
+
+  toggleDeleteDialog = (isDeleteDialog: boolean) => {
+    this.setState({
+      isDeleteDialog,
+    })
+  }
+
+  render() {
+    const {
+      deleteAssetItem,
+      name,
+      symbol,
+      balance,
+      isCustom,
+    } = this.props
+
+    const {
+      isChange,
+      isHoveredEdit,
+      isHoveredTrash,
+      isDeleteDialog,
+    }: StateProps = this.state
+
+    return (
+      <JCard color='white' isBorderRadius isHover>
+        <div className='asset-item'>
+          <div className='info'>
+            {!isCustom ? (
+              <div className='symbol -icon'>
+                <JAssetSymbol symbol={symbol} color='gray' />
+              </div>
+            ) : (
+              <div className='symbol -text'>
+                <JText value={symbol} color='blue' weight='bold' size='normal' whiteSpace='wrap' />
+              </div>
+            )}
+            <div className='data'>
+              <div className='name'>
+                <JText value={name} color='dark' weight='bold' size='normal' whiteSpace='wrap' />
+              </div>
+              <div className='balance'>
+                <AssetBalance
+                  color='gray'
+                  size='small'
+                  symbol={symbol}
+                  balance={balance}
+                />
+              </div>
+            </div>
+          </div>
+          <div className='actions'>
+            {isCustom && (
+              <Fragment>
+                {!isDeleteDialog ? (
+                  <Fragment>
+                    <div
+                      className='overlay'
+                      onClick={handle(this.toggleChange)(!isChange)}
```

Comment:
this event should be fired by click on overlay only when `isChange === true`
another way is hide overlay (to make it unclickable)